tests/integration/modifiers/did-intersect-test.js

```diff
@@ -1,87 +1,97 @@
 import { module, test } from 'qunit';
 import { setupRenderingTest } from 'ember-qunit';
-import { render } from '@ember/test-helpers';
+import { render, setupOnerror } from '@ember/test-helpers';
 import hbs from 'htmlbars-inline-precompile';
 import sinon from 'sinon';
-
-let intersectionCallback;
-let intersectionOptions;
-let observeStub;
-let disconnectStub;
-let MockIntersectionObserver;
+import Service from '@ember/service';
 
 module('Integration | Modifier | did-intersect', function (hooks) {
   setupRenderingTest(hooks);
 
-  let intersectionObserver;
-
   hooks.beforeEach(function () {
-    intersectionCallback = null;
-    intersectionOptions = null;
-    observeStub = sinon.stub();
-    disconnectStub = sinon.stub();
-
-    MockIntersectionObserver = class MockIntersectionObserver {
-      constructor(callback, options) {
-        intersectionCallback = callback;
-        intersectionOptions = options;
+    class MockObserverManager extends Service {
+      constructor() {
+        super(...arguments);
+        this._admin = {};
       }
 
-      observe = observeStub;
-      disconnect = disconnectStub;
+      observe = sinon.stub();
+      unobserve = sinon.stub();
+      addEnterCallback = sinon.stub();
+      addExitCallback = sinon.stub();
     }
 
-    intersectionObserver = window.IntersectionObserver;
-    window.IntersectionObserver = MockIntersectionObserver;
+    this.owner.register('service:ember-scroll-modifiers@observer-manager', MockObserverManager);
 
-    this.intersectStub = sinon.stub();
+    this.observerManagerMock = this.owner.lookup('service:ember-scroll-modifiers@observer-manager');
+    this.enterStub = sinon.stub();
+    this.exitStub = sinon.stub();
   });
 
-  hooks.afterEach(function () {
-    window.IntersectionObserver = intersectionObserver;
-  });
+  test('modifier integrates with observer-manager and triggers correct callbacks when onEnter and onExit are provided', async function (assert) {
+    assert.expect(3);
```

Comment:
We don't need `assert.expect` for any of these tests except where there are callbacks executing the assertions. Thanks to async/await, these assertions are all run linearly and so we don't need to assert that the correct number of assertions occurred. Go ahead and remove these so we don't have to update them as we add/remove assertions in the future.